deploy-agent/deployd/client/client.py

```diff
@@ -171,11 +172,22 @@ def _read_host_info(self):
             # so need to read ec2_tags from facter and parse Autoscaling tag to cover this case
             if not self._autoscaling_group:
                 ec2_tags = facter_data.get(ec2_tags_key)
-                if ec2_tags:
-                    ec2_tags['availability_zone'] = self._availability_zone
-                self._ec2_tags = json.dumps(ec2_tags) if ec2_tags else None
                 self._autoscaling_group = ec2_tags.get(asg_tag_key) if ec2_tags else None
 
+            # Obtain all EC2 tags, inclusive of those that are outside the scope of the 'facter' command.
+            self._ec2_tags = None
+            try:
+                ec2 = boto3.resource('ec2', region_name='us-east-1')
```

Comment:
I will cause additional aws requests right? do we currently fetch these tags from cmdb?

Reply:
Yes, it does invoke AWS. We need to retrieve all tags from AWS and then save them in our database. Without this step, a tag added by a user cannot be fetched through the 'facter' command. As of now, all tags are stored in our database by querying them from the CMDB.